### PR TITLE
fix: SRS-591-backend

### PR DESCRIPTION
**Pull request by jaspalsingh-aot**
# Description

This PR implements global error handling to enhance the application's resilience and user experience. The changes include:

- Global Error Handling: All HTTP exceptions are now caught, allowing for a standardized response.
- Internal Server Error Handling: Other exceptions will return a consistent internal server error message.
- Error Response Format: A uniform format for error responses has been established for better clarity.

Screenshot 2024-10-24 111349

Screenshot 2024-10-24 100836

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-132-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-132-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/app/filters/customExceptionFilters.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Catch, ArgumentsHost, HttpException } from '@nestjs/common';
+import { GqlExceptionFilter } from '@nestjs/graphql';
+import { GqlArgumentsHost } from '@nestjs/graphql';
+import { GenericResponse } from '../dto/response/genericResponse';
+
+@Catch()
+export class CustomExceptionFilter implements GqlExceptionFilter {
+  catch(exception: HttpException, host: ArgumentsHost) {
+    const status =
+      exception instanceof HttpException ? exception.getStatus() : 500;
```
Comment (nikhila-aot) [inline backend/src/app/filters/customExceptionFilters.ts]: Can we have all numeric error codes replaced with Http error codes

Comment (jaspalsingh-aot) [reply backend/src/app/filters/customExceptionFilters.ts]: Done

**Review by nikhila-aot — DISMISSED** [review]

File: backend/src/app/filters/customExceptionFilters.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import {
+  Catch,
+  ArgumentsHost,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { GqlExceptionFilter } from '@nestjs/graphql';
+import { GqlArgumentsHost } from '@nestjs/graphql';
```
Comment (mishraomp) [inline backend/src/app/filters/customExceptionFilters.ts]: unused import

File: backend/src/app/filters/customExceptionFilters.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { GenericResponse } from '../dto/response/genericResponse';
+
+@Catch()
+export class CustomExceptionFilter implements GqlExceptionFilter {
+  catch(exception: HttpException, host: ArgumentsHost) {
```
Comment (mishraomp) [inline backend/src/app/filters/customExceptionFilters.ts]: unused param host

File: backend/src/app/filters/customExceptionFilters.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        ? exception.getResponse()
+        : 'Internal server error';
+    const message =
+      typeof messageResponse === 'string'
+        ? messageResponse
+        : messageResponse['message'];
+
+    // Format the error response
+    const result = new GenericResponse(message, status, false, null);
+    return result;
```
Comment (mishraomp) [inline backend/src/app/filters/customExceptionFilters.ts]: return without const, `return new GenericResponse(message, status, false, null);`

File: backend/src/app/resolvers/dropdown/dropdown.resolver.ts [hunk 5]
```diff
@@ -75,35 +76,23 @@ export class DropdownResolver {
...
-      }
-    } catch (error) {
-      this.sitesLogger.error(
-        'Exception occured in DropdownResolver.getPeopleOrgsCd() end',
-        JSON.stringify(error),
+    const result = await this.dropdownService.getPeopleOrgsCd(
+      searchParam,
+      entityType,
+    );
+    if (result && result.length > 0) {
```
Comment (mishraomp) [inline backend/src/app/resolvers/dropdown/dropdown.resolver.ts]: please use `result?.length >0`

File: backend/src/app/resolvers/folio/folio.resolver.ts [hunk 6]
```diff
@@ -40,36 +40,24 @@ export class FolioResolver {
...
-          404,
-          true,
-          [],
-        );
-      }
-    } catch (error) {
-      this.sitesLogger.error(
-        'Exception occured in FolioResolver.getFolioItemsForUser() end',
-        JSON.stringify(error),
+    if (result && result.length > 0) {
```
Comment (mishraomp) [inline backend/src/app/resolvers/folio/folio.resolver.ts]: please use optional chaining

File: backend/src/app/resolvers/snapshot/snapshot.resolver.ts [hunk 7]
```diff
@@ -193,44 +193,39 @@ export class SnapshotsResolver {
         ' ' +
         JSON.stringify(inputDto),
     );
-    try {
-      if (inputDto) {
-        const isSaved = this.snapshotsService.createSnapshotForSites(
-          inputDto,
-          user,
+    if (inputDto) {
+      const isSaved = this.snapshotsService.createSnapshotForSites(
```
Comment (mishraomp) [inline backend/src/app/resolvers/snapshot/snapshot.resolver.ts]: missing `await`

File: backend/src/app/resolvers/snapshot/snapshot.resolver.ts [hunk 8]
```diff
@@ -249,35 +244,27 @@ export class SnapshotsResolver {
...
-          data: null,
-        };
-      }
-    } catch (error) {
-      this.sitesLogger.error(
-        'Exception occured in SnapshotsResolver.getBannerType() end',
-        JSON.stringify(error),
-      );
-      throw new Error('System Error, Please try again.');
+    if (bannerType && bannerType.length > 0) {
```
Comment (mishraomp) [inline backend/src/app/resolvers/snapshot/snapshot.resolver.ts]: use optional chaining

**Review by mishraomp — CHANGES_REQUESTED** [review]
really great work, very few minor comments

File: backend/src/app/filters/customExceptionFilters.ts (hunk 2 above)
Comment (jaspalsingh-aot) [reply backend/src/app/filters/customExceptionFilters.ts]: done

File: backend/src/app/filters/customExceptionFilters.ts (hunk 4 above)
Comment (jaspalsingh-aot) [reply backend/src/app/filters/customExceptionFilters.ts]: done

File: backend/src/app/resolvers/dropdown/dropdown.resolver.ts (hunk 5 above)
Comment (jaspalsingh-aot) [reply backend/src/app/resolvers/dropdown/dropdown.resolver.ts]: done

File: backend/src/app/resolvers/folio/folio.resolver.ts (hunk 6 above)
Comment (jaspalsingh-aot) [reply backend/src/app/resolvers/folio/folio.resolver.ts]: done

File: backend/src/app/resolvers/snapshot/snapshot.resolver.ts (hunk 7 above)
Comment (jaspalsingh-aot) [reply backend/src/app/resolvers/snapshot/snapshot.resolver.ts]: done

File: backend/src/app/resolvers/snapshot/snapshot.resolver.ts (hunk 8 above)
Comment (jaspalsingh-aot) [reply backend/src/app/resolvers/snapshot/snapshot.resolver.ts]: done

**Review by mishraomp — APPROVED** [review]
nice work, LGTM

**Pull request merged**